### PR TITLE
Remove deprecated summary fields totalCellCount and projectEstimatedCellCount (#3650)

### PR DESCRIPTION
**Pull request by danielsotirhos**
Connected issue: #3650

### Author

- [x] PR title references the connected issue
- [x] PR title matches<sup>1</sup> title of connected issue         <sub>or comment in PR explains why they're different</sub>
- [x] Title of at least one commit references connected issue
- [x] PR is connected to issue via Zenhub 
- [x] PR description links to connected issue

<sup>1</sup> when the issue title describes a problem, the PR title is `Fix: ` followed by the issue title   

### Author (reindex)

- [x] Added `r` tag to commit title                                 <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                                   <sub>or this PR does not require reindexing</sub>

### Author (chains)

- [x] This PR is blocked by previous PR in the chain                <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                        <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst          <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                                 <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                                   <sub>or this PR does not require upgrading</sub>
- [x] Added announcement to PR description                          <sub>or this PR does not require announcement</sub>
- [x] Added checklist items for additional operator tasks           <sub>or this PR does not require additional tasks</sub>

## Author (hotfixes)

- [x] Added `h` tag to commit title and PR targets `prod`           <sub>or this PR does not include a temporary hotfix</sub>
- [x] Added `H` tag to commit title                                 <sub>or this PR does not include a permanent hotfix</sub>
- [x] Added `hotfix` label to PR                                    <sub>or this PR does not include a hotfix</sub>
- [x] Reverted the temporary hotfix connected to the issue          <sub>or there is no temporary hotfix for the connected issue on the `prod` branch</sub>

### Author (requirements, before every review)

- [x] Ran `make requirements_update`                                <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title                                 <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR                                      <sub>or this PR does not touch requirements*.txt</sub>

### Author (before every review)

- [x] `make integration_test` passes in personal deployment         <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

### Primary reviewer (after approval)

- [x] Actually approved the PR
- [x] Labeled connected issue as `demo` or `no demo`
- [x] Commented on connected issue about demo expectations          <sub>or labelled connected issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Ticket is in *Review requested* column
- [x] Requested review from peer reviewer
- [x] Assigned PR to peer reviewer

### Peer reviewer (after approval)

- [x] `N reviews` label on PR reflects prior peer reviews with changes requested
- [x] Moved ticket to *Approved* column
- [x] Assigned PR to current operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear                      <sub>or connected issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab and added `sandbox` label             <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passed in sandbox                                       <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                                  <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                             <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved connected issue to Merged column
- [ ] Pushed merge commit to Github

### Operator (after pushing the merge commit)

- [ ] Made announcement requested by author                         <sub>or PR description does not contain an announcement</sub>
- [ ] Shortened the PR chain                                        <sub>or this PR is not the base of another PR</sub>
- [ ] Pushed merge commit to Gitlab                                 <sub>or merge commit can be pushed later, with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Build passes on Gitlab
- [ ] Moved connected issue to `prod` or `Merged prod`              <sub>or this PR does not represent a promotion</sub>

### Operator (reindex) 

- [ ] Started reindex in target deployment                          <sub>or this PR does not require reindexing</sub>
- [ ] Checked for and triaged indexing failures                     <sub>or this PR does not require reindexing</sub>
- [ ] Emptied fail queues in target deployment                      <sub>or this PR does not require reindexing</sub>
- [ ] Filed backport PR                                             <sub>or this PR does not represent a hotfix</sub>

### Operator

- [ ] Unassigned PR

## Shorthand for review comments

- `L` line is too long
- `W` line wrapping is wrong
- `Q` bad quotes
- `F` other formatting problem

### CODE REVIEW

**Review by hannes-ucsc — APPROVED** [review]

**Review by noah-aviel-dove — CHANGES_REQUESTED** [review]
There is a lingering reference to `totalCellCount` in the Swagger documentation.

```diff
Index: lambdas/service/app.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lambdas/service/app.py b/lambdas/service/app.py
--- a/lambdas/service/app.py	(revision 7479985585769c60003df7f59a2c3537df902c74)
+++ b/lambdas/service/app.py	(date 1648544706921)
@@ -1356,7 +1356,6 @@
                 `fileCount` and `totalFileSize` compile these figures across all
                 file formats. Likewise, `cellCountSummaries` counts cells and
                 their associated documents grouped by organ type, with
-                `totalCellCount` compiling cell counts across organ types and
                 `organTypes` listing all referenced organs.
 
                 Total counts of unique entities are also provided for other
```

**Review by noah-aviel-dove — CHANGES_REQUESTED** [review]
It seems that the `projects` response field wasn't properly documented, and that this represents a documentation bug that's unrelated to the removal of the deprecated fields. I think those changes should be moved to a separate commit that precedes the main commit.

I suggest you re-request review from @hannes-ucsc after making those changes, since this PR now makes user-facing changes to the documentation that he hasn't had a chance to review.

File: lambdas/service/app.py [hunk 1]
```diff
@@ -1354,10 +1354,14 @@ def get_project_data(project_id: Optional[str] = None) -> JSON:
...
                 `fileCount` and `totalFileSize` compile these figures across all
-                file formats. Likewise, `cellCountSummaries` counts cells and
-                their associated documents grouped by organ type, with
-                `totalCellCount` compiling cell counts across organ types and
-                `organTypes` listing all referenced organs.
+                file formats.
+
+                `cellCountSummaries` provides cell suspension cell counts and
+                the number of associated documents grouped by organ type, with
+                `organTypes` listing all referenced organs. `projects`
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:1361]: `projects` will have other summaries as well, not just cell counts. The structure is generally `outer_entity.inner_entity.property`.

I don't want to get hung up on this right now, just remove the change and file a FIXME for #3917, please.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: lambdas/service/app.py (hunk 1 above)
Comment (danielsotirhos) [reply lambdas/service/app.py]: Created https://github.com/DataBiosphere/azul/issues/4016 to address the missing API documentation of the summary response field `projects`.

Comment (hannes-ucsc) [reply lambdas/service/app.py]: I meant to ask you to add a FIXME referring to #3917. No new issue is needed.

**Review by hannes-ucsc — APPROVED** [review]

**Review by noah-aviel-dove — APPROVED** [review]

**Pull request merged**